tiledb/cc/array.cc

```diff
@@ -55,17 +55,57 @@ void init_array(py::module &m) {
       .def("config", &Array::config)
       .def("close", &Array::close)
       .def("consolidate",
-           py::overload_cast<const Context &, const std::string &,
-                             Config *const>(&Array::consolidate),
-           py::call_guard<py::gil_scoped_release>())
+           [](Array &self) {
+             if (self.query_type() == TILEDB_READ) {
+               throw TileDBError("cannot consolidate array opened in readonly "
+                                 "mode (mode='r')");
+             }
+
+             Config config;
+             Context ctx;
+             Array::consolidate(ctx, self.uri(), &config);
+           })
```

Comment:
I don't think we should have this one. The default context should be coming from the already existing `tiledb.default_ctx()` rather than constructing a new `tiledb::Context`.

Reply:
But if we delete this, we cannot have calls like `array.consolidate()` without any arguments.
This can be done with current implementation.